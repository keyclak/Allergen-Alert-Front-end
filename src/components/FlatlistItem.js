import React, {useState} from 'react'
import { View, Text, StyleSheet, Pressable, Image} from 'react-native';
import { ListItem, Icon } from 'react-native-elements'
import { StyleConstants, Styles, Colors } from '../style';

export default function FlatlistItem(props) {

    function handlePress(location) {
        console.log(location)
    }

    return(
        <ListItem key={props.i} bottomDivider onPress={() => {handlePress(props.item.navTo)}}>
            <Icon name={props.item.icon} color={"#666"}/>
            <ListItem.Content>
            <ListItem.Title style={[Styles.labelText, {color:"#666"}]}>{props.item.title}</ListItem.Title>
            </ListItem.Content>
            <ListItem.Chevron color="#666"/>
        </ListItem>
    )
}