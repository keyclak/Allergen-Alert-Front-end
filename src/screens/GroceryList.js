import React, {useEffect, useState} from 'react'
import {SafeAreaView, ScrollView, View, Text, FlatList, Pressable} from 'react-native'
import { StyleConstants, Styles, Colors } from '../style';
import { useDeleteGrocery, useGetFoodByID, useGetGroceryList, useTogglePurchased } from '../hooks/api';
import { ListItem, CheckBox } from 'react-native-elements'

export default function GroceryList({navigation}) {

    const getGroceryList = useGetGroceryList();
    const getFoodByID = useGetFoodByID(); 
    const deleteGrocery = useDeleteGrocery(); 
    const togglePurchased = useTogglePurchased(); 

    useEffect(() => {
        navigation.addListener('focus', () =>  getGroceryList.background());
    }, [navigation]);

    function onDelete(id) {
        deleteGrocery.execute(id)
            .then(() => getGroceryList.background())
            .catch(e => {});
    }

    function onCheck(food) {
        togglePurchased.execute(food.foodId, !food.purchased)
            .then(() => getGroceryList.background())
            .catch(e => {});    
    }

    return (
        <SafeAreaView style={[Styles.container, {flex: 1, alignItems: 'flex-start', justifyContent:'space-evenly'}]}>
            <View style={{paddingTop: 15}}>
                <FlatList
                    data={getGroceryList.response}
                    renderItem={({ item }) => (
                        <ListItem containerStyle={{backgroundColor: Colors.Background, padding: 2}}>
                            <CheckBox
                                containerStyle={{width: '72%', borderRadius: 15}}
                                // TODO: use item.name
                                title={item.foodId}
                                checked={item.purchased}
                                onIconPress={() => onCheck(item)}
                            />
                            <Pressable 
                                style= {{backgroundColor: Colors.ButtonBackground, padding: 15, borderRadius: 25}}
                                onPress={() => onDelete(item.foodId)}>
                                <Text style={{color: Colors.Accent}}>Delete</Text>
                            </Pressable>
                        </ListItem>
                    )}
                    keyExtractor={item => `${item.foodId}`}
                />
            </View>
        </SafeAreaView>
    )
}