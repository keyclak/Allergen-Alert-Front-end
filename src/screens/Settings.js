import React, { useContext, useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet} from 'react-native';
import LoadingButton from '../components/LoadingButton';
import TextLoadingButton from '../components/TextLoadingButton';
import { Styles } from '../style';
import { useLogin } from '../hooks/api';
import { AuthContext } from '../context';
import { ListItem, Icon } from 'react-native-elements'

export default function Settings() {
    const context = useContext(AuthContext);

    return (
        <View style={[Styles.container, {backgroundColor: 'white'}]}>   
            {/* <TextLoadingButton text="Log Out" onPress={() => context.logOut()} /> */}
            <ListItem style = {{width: '100%'}} bottomDivider onPress={() => context.logOut()}>
                <Icon name='keyboard-tab' color={"#666"}/>
                <ListItem.Title style={{color:"#666"}}>
                    Log Out
                </ListItem.Title>
            </ListItem>
        </View>
    );
}