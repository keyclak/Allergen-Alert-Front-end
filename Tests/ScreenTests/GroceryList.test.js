import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import FlatlistItem from '../../src/components/FlatlistItem';
import { create } from "react-test-renderer";
import { FlatList } from 'react-native-gesture-handler';
import { useNavigation } from '@react-navigation/native';
import renderer from 'react-test-renderer';
import Settings from '../../src/screens/Settings';
import GroceryList from '../../src/screens/GroceryList';

describe("GrocerList tests", () => {
    test("Renders correctly", () => {
        const tree = renderer.create(<GroceryList></GroceryList>);

        expect(tree.toJSON()).toMatchSnapshot();
    })
})