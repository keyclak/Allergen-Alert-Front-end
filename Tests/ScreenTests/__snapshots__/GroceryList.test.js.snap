// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrocerList tests Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      },
      Object {
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    data={null}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
